File: C_Programming/Practise_Programs/PP_182.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_182
//
//  Description       : This program toggles the 12th and 23rd bits of
//                      a given unsigned integer using the bitwise
//                      XOR operator and an appropriate bit mask.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 23/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

// Position 12th and 23rd
int main()
{
    UINT iMask = 0x00400800;
    UINT iNo = 0;
    UINT iResult = 0;

    printf("Enter Number : \n");
    scanf("%u", &iNo);

    iResult = iNo ^ iMask;

    printf("Updated No : %u\n", iResult);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//
//  Sample Output :
//
//      Enter Number :
//      Updated No : 4196362
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_182.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_182
//
//  Description       : This program accepts three numbers from the
//                      user and finds the smallest among them.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 07/09/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_182
{
    public static int FindSmall(int iNo1, int iNo2, int iNo3)
    {
        if(iNo1 < iNo2 && iNo1 < iNo3)
        {
            return iNo1;
        }
        else if(iNo2 < iNo1 && iNo2 < iNo3)
        {
            return iNo2;
        }
        else 
        {
            return iNo3;
        }
    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0, iValue3 = 0, iRet = 0;

        System.out.println("Enter the First Number : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the Second Number : ");
        iValue2 = sobj.nextInt();

        System.out.println("Enter the Third Number : ");
        iValue3 = sobj.nextInt();

        iRet = FindSmall(iValue1, iValue2, iValue3);

        System.out.println("Smallest Number is : " + iRet);

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the First Number :
//      25
//      Enter the Second Number :
//      78
//      Enter the Third Number :
//      42
//
//  Sample Output :
//      Smallest Number is : 25
//
//////////////////////////////////////////////////////////////////